eq-author/src/components/QuestionPicker/index.js

```diff
@@ -0,0 +1,282 @@
+    if (itemSelected) {
+      const selectionWithoutThisPage = selectedPages.filter(
+        (selectedPage) => selectedPage.id !== page.id
+      );
+      updateSelectedPages(selectionWithoutThisPage);
+    } else {
+      updateSelectedPages([...selectedPages, page]);
+    }
+  };
+  return (
+    <Item
+      title={title.replace(/(<([^>]+)>)/gi, "") || displayName}
+      subtitle={alias}
+      onClick={handleClick}
+      selected={itemSelected}
+      dataTest="Page"
+    />
+  );
+};
+Page.propTypes = {
+  page: PropTypes.object, // eslint-disable-line
+};
+
+const Folder = ({ folder }) => {
+  const { displayName, pages } = folder;
+
+  const numOfPagesInFolder = pages.flatMap(({ pages }) => pages).length;
+
+  if (numOfPagesInFolder > 0) {
+    return (
+      <Item
+        icon={<FolderIcon />}
+        title={displayName}
+        unselectable
+        dataTest="folder"
+      >
+        <List className="sublist">
+          {pages.map((page) => (
+            <Page key={`page-${page.id}`} page={page} />
+          ))}
+        </List>
+      </Item>
+    );
+  }
+
+  return <React.Fragment />;
+};
+Folder.propTypes = {
+  folder: PropTypes.object, // eslint-disable-line
+};
+
+const Section = ({ section }) => {
+  const { displayName, folders } = section;
+
+  const numOfPagesInSection = folders.flatMap(({ pages }) =>
+    pages.flatMap(({ pages }) => pages)
+  ).length;
+
+  if (numOfPagesInSection > 0) {
+    return (
+      <Item variant="heading" title={displayName} unselectable>
+        <List>
+          {folders.map((folder) => {
+            const { enabled } = folder;
+            if (enabled) {
+              return <Folder key={`folder-${folder.id}`} folder={folder} />;
+            } else {
+              return folder.pages.map((page) => (
+                <Page key={`page-${page.id}`} page={page} />
+              ));
+            }
+          })}
+        </List>
+      </Item>
+    );
+  }
+
+  return <React.Fragment />;
+};
+Section.propTypes = {
+  section: PropTypes.object, // eslint-disable-line
+};
+
+const QuestionPicker = ({
+  title,
+  sections,
+  warningPanel,
+  showSearch,
+  isOpen,
+  onClose,
+  onSubmit,
+  startingSelectedQuestions = [],
+}) => {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [filteredSections, updateFilteredSections] = useState([]);
+  const [selectedPages, updateSelectedPages] = useState([]);
+
+  useEffect(() => {
+    updateFilteredSections(filterList(sections, searchTerm));
+  }, [sections, searchTerm]);
+
+  useEffect(() => {
+    updateSelectedPages(startingSelectedQuestions);
+  }, [startingSelectedQuestions]);
+
+
+  const filterList = (data, searchTerm) => {
+    const lowerCaseSearchTerm = searchTerm.toLowerCase();
+
+    return data.map(({ folders, ...rest }) => ({
+      folders: folders.map(({ pages, ...rest }) => ({
+        pages: pages.filter(({ displayName }) =>
+        displayName.toLowerCase().includes(lowerCaseSearchTerm)
```

Comment:
Hi Tucker,

Yup - so regarding the searching issue, here's the data available regarding naming of questions:

------

#### `alias`
The "short name" shown in eQ Author, may or may not be present

#### `title`
The "title" shown in eQ Author

#### `displayName`

The way eQ Author will show the item in the nav bar, equivalent  to:

```javascript
alias || title || "Untitled question"
```

-----

Probably what you want to do to make sure you can search the short code and the title in one go is either to test both against the string or concatenate them and then test once, e.g.:

```suggestion
        pages: pages.filter(({ alias, title, displayName }) =>
            `${alias ? alias : ""} ${title ? title : displayName}`.toLowerCase().includes(lowerCaseSearchTerm)
```

Reply:
Implemented this change as this solved the issue.